### ISSUE TITLE
Delivered Status Gone After Update 2024-04-11-production.0

### ISSUE THREAD

**Issue by Sledge-Hammer** [wireapp/wire-webapp#17276]
Hello,

Before Update 2024-04-11-production.0 there was a "Delivered" status next to each message and now it's gone.

There's nothing in the changelog about it, so I'm guessing that it's a bug (probably related to the new 100% width of the messages) or it was moved to the "Read Receipts" code, which works only if activated by each contact.

In either case - please bring it back, because this is a complete downgrade. 
If it has become part of the "Read Receipts" - that is not the right place for it.

I do not really care if someone has read a message or when they've read it (in case such info is provided (I've never used this option)), but I do really care whether the message has been delivered or something happened during delivery.

For example I'm often sending messages that do not require an answer, but I still need to know that the other party has received them (and search for other means of communication with them if they haven't received them after a certain period).

It's not realistic to ask each of my contacts to enable read receipts and if you're worried about spamming - make the "Delivered" status available only for people in our contacts list.

In case it matters - this happens both in a browser and the latest for the moment version of the desktop client.

**Comment by atomrc** [comment]
Hi @Sledge-Hammer,

Thanks for your report. 
Something I'd like to confirm with you. The main change around delivery receipts are that they are only shown on the _last delivered message_ and not the ones before that. 

Could this be what misled you to think that the `Delivered` status was not there at all?

In case you feel the status is really missing, could you give us a few more information:
- is it happening in a 1 to 1 conversation or a group conversation
- are you part of a team or have a private account?

Thanks again

**Comment by Sledge-Hammer** [comment]
@atomrc Sorry, I've explained the issue poorly.
The "Delivered" status was visible only next to the last delivered message, not next to every message in the chat.

Before the last update this feature was working only partially - the "Delivered" status was never displayed while I was talking to people using the new Android app, but now it's gone from all conversations.

I've asked a few of my contacts and all of them confirmed the same thing.

This happens in a 1 to 1 conversations, I have no idea about group chats, because I'm not using any at the moment.
Mine is a private account.

Thank you for taking a look at this and please let me know if I can provide any additional info.

**Comment by tlebon** [comment]
I did a small investigation on this topic today. 

"Delivered" will only appear on the lowest delivered message in the chat if you are hovering over it. 
Additionally, the user has to have a client online for the message to be delivered. 
If read receipts are turned on then the read receipt will replace the delivered message when the message is read (any read message has already been delivered as well)

In testing this we did discover that between android and webapp there appears to be an issue with the delivered status propagation, so we have created a ticket for the android team to investigate this. We will follow up when this is fixed/ a timeline for that.

**Comment by Sledge-Hammer** [comment]
@tlebon Please, turn off read receipts for your personal account (I have no idea if this bug exists for business accounts) and test again, because I can assure you that the only thing visible on hover is the emoji menu.

Just to be sure that I'm not missing something obvious, I turned on read receipts and the "Delivered" message did show up next to the last message on hover.

And it was completely gone as soon as I turned off read receipts again, so it's not just the Android bug, you guys should also make things work with read receipts being turned off (as it was before update 2024-04-11-production.0).

It would be really appreciated if you can also make it visible without the need to hover (for example if I've sent 10 messages to someone and they've received only the first 5 I'll have to hover over each message to find out the last received instead of just seeing it as soon as I click on that chat) and also put it back on the right of the message (below the emoji menu). Basically it was perfect the way it was before the update and the new changes feel like a downgrade.

**Comment by tlebon** [comment]
ill take a look at this also.

**Comment by tlebon** [comment]
Ok you are correct, this is a regression and appears to be tied to read-receipts accidentally. I am unsure if the delivered status will *not* be on hover (as this was a design decision) but we will open a ticket for this issue.

**Comment by tlebon** [comment]
ticket is open, seems we have a solution, should be included in the next release, hopefully in a few weeks.

**Comment by Sledge-Hammer** [comment]
@tlebon Thank you!

As for the hover issue - could you please ask the person responsible for that design decision to take a look at this topic.

The updated design might look cleaner but it's bad user experience and it creates 2 new issues.
&nbsp;

The first one is the need to hover on each message to find the last delivered one.
I completely understand changing the read receipts to be visible only on hover since they're available for each message, but the delivered status is available only for the last received message and should be always visible, instead of making us hunt for it.

&nbsp;
The second one is the new position at the end of each message.
This is bad both for the read receipts and the delivered status because each message has different length and people have to constantly scan with their eyes for the end of the message to see if the "Delivered" status is there (wastes time and requires extra eye movements) or in the case of the read receipts they also have to move the mouse cursor to the end of each message if they want to see the time the message was read (if everything is on the right of the window the mouse can be moved just up and down).
&nbsp;

In summary for the best user experience - both the read icon and the delivered status should be located at the right side of the window, below the emoji menu and the Delivered status should be always visible while the read icon should be left hidden and shown only on hover.
Also if the user has read receipts enabled you could make the read icon visible without hover for the last read message if it's also the last delivered message.

&nbsp;
If not - at the very least give us an option to add custom CSS rules that won't be overridden on app update.

**Comment by tlebon** [comment]
i passed this along to design as well, we can update with their feedback on the topic.

**Comment by Sledge-Hammer** [comment]
Any chance of this ever getting fixed?
I tried the latest version from today (Wire for web version 2024.08.07.10.06.46) and the Delivered status is still displayed only when read receipts are enabled.

**Comment by Sledge-Hammer** [comment]
@tlebon On top of the still not fixed "Delivered" issue I just noticed a new bug related to the read receipts.

Shortly after my last message here (after I had already updated my desktop client) I was talking to someone in Wire with read receipts enabled on both their desktop and phone (android), while mine are off everywhere (active conversation with them replying as soon as I sent a message and not in a group chat). While I was talking to that person they were using the phone while being outside, but their desktop was also online at the same time.
During our conversation I saw nothing unusual, but a few hours later they sent me a message from their desktop with the still active older version of Wire.
After that I started seeing read receipts on hover for every message I've sent them up until May this year (in other words probably everything between today and the previous Wire update).
A few minutes later they updated their desktop version and I stopped seeing receipts for the new messages I've sent, but the ones that were already revealed are still visible even after both of us restarted Wire.

Not sure you'll be able to replicate it, but I'm sure that someone else out there had the same issue.

I guess it's not that big of a problem considering that they had the receipts enabled, though I'm still not supposed to see them since mine are disabled.
